.github/workflows/pr-trigger.yml

```diff
@@ -20,14 +20,22 @@ jobs:
         with:
           fetch-depth: 0
 
+      - name: Check node version
+        id: node-version
+        run: |
+          npmVersion=$(cat .nvmrc | head -n 1)
+          echo ::set-output name=npmVersion::${npmVersion}
+
       - name: Setup node
         uses: actions/setup-node@v3
+        with:
+          node-version: ${{ steps.node-version.outputs.npmVersion }}
 
-      - name: Remove package.json & yarn.lock to avoid installing everything
-        run: rm package.json yarn.lock
-
-      - name: Install nx
-        run: yarn add nx
+      - name: Setup nx
+        run: |
+          rm package.json yarn.lock
+          yarn add nx
+          git restore package.json yarn.lock
```

Comment:
**thought (non-blocking)**: An alternative to this would be to use `npx`, which is like `npm` but executes it. Something like:

```bash
npx  --silent --yes nx affected...
```

Should be equivalent to this. I'm not sure if the package.json version of `nx` is significant or not (i.e. if we should make sure we are installing the same major version of `nx`), but `npx` would handled that with `npx nx@17` or a similar semver string

Reply:
nice one! added

Reply:
At the end I had issues with making this work, but I have used your suggestion to use pinned version of `nx` to make sure we use the same one